mmedit/models/restorers/basicvsr.py

```diff
@@ -0,0 +1,179 @@
+import mmcv
+import numbers
+import numpy as np
+import os.path as osp
+import torch
+
+from mmedit.core import tensor2img
+from ..registry import MODELS
+from .basic_restorer import BasicRestorer
+
+
+@MODELS.register_module()
+class BasicVSR(BasicRestorer):
+    """BasicVSR model for video super-resolution.
+
+    Note that this model is used for IconVSR.
+
+    Paper:
+        BasicVSR: The Search for Essential Components in Video Super-Resolution
+        and Beyond, CVPR, 2021
+
+    Args:
+        generator (dict): Config for the generator structure.
+        pixel_loss (dict): Config for pixel-wise loss.
+        train_cfg (dict): Config for training. Default: None.
+        test_cfg (dict): Config for testing. Default: None.
+        pretrained (str): Path for pretrained model. Default: None.
+    """
+
+    def __init__(self,
+                 generator,
+                 pixel_loss,
+                 train_cfg=None,
+                 test_cfg=None,
+                 pretrained=None):
+        super(BasicVSR, self).__init__(generator, pixel_loss, train_cfg,
+                                       test_cfg, pretrained)
+
+        # fix pre-trained networks
+        self.fix_iter = train_cfg.get('fix_iter', 0) if train_cfg else 0
+        self.generator.find_unused_parameters = False
+
+        self.step_counter = 0  # count training steps
```

Comment:
Does this parameter (`step_counter `) need to be changed when model is resumed?

Reply:
Yes. It should start with the resume iteration. For instance, if we resume at 10000-th iter, we should have `self.step_counter=10000`. 

Maybe we can pass an argument in `train_cfg`? Or is there any better way to do that?

Reply:
Passing the value from `train_cfg` is a good way.

Reply:
Passing the value from A is a good idea.

Reply:
Passing the value from `train_cfg` is a good idea.